pkg/rpc/client/rpc.go

```diff
@@ -453,11 +450,13 @@ func (c *Client) SignAndPushInvocationTx(script []byte, acc *wallet.Account, sys
 		return txHash, errors.Wrap(err, "failed to sign tx")
 	}
 	txHash = tx.Hash()
-	err = c.SendRawTransaction(tx)
-
+	res, err := c.SendRawTransaction(tx)
 	if err != nil {
 		return txHash, errors.Wrap(err, "failed sendning tx")
 	}
+	if !res.Hash.Equals(txHash) {
```

Comment:
When is this true? Assuming non-malicious node of course.

Reply:
Removed this check.